nengo/simulator.py

```diff
@@ -55,11 +55,12 @@ def __len__(self):
 class Simulator(object):
     """Reference simulator for Nengo models."""
 
+    # 'unsupported' defines features unsupported by a simulator.
+    # The format is a list of tuples of the form `(test, reason)` with `test`
+    # being a regular expression matched against Nengo test names, and `reason`
+    # is a string describing why the feature is not supported by the backend.
+    # E.g., unsupported = [('test_pes*', 'PES rule not implemented')]
```

Comment:
One more question about this... we match on the `nodeid`, which includes the path (an example nodeid is `nengo/tests/test_learning_rules.py::test_pes_weight_solver`). This is good, I think, as then we can xfail on files in addition to test names. However, things like `test_pes*` won't work with `fnmatch` because the path comes before `test_pes` in the nodeid.

I can think of a few options in response to this:

1. Keep it as is, give an example nodeid in the comment for `unsupported`, and make the example `*test_pes*` instead of `test_pes*`.
2. Automatically add `*` to the start and end of the test strings in `unsupported` (e.g., `test_pes*` is interpreted the same as `test_pes` which is the same as `*test_pes`).
3. Instead of wildcard matching, just do a substring test (i.e., `test in item.nodeid`). This is what `py.test -k` does (sort of; `-k` allows multiple substrings with boolean operators, so something like 'test_pes or test_unsupervised` would match both).

Any opinions? I'm not sure I have a strong preference between the three choices...